src/components/button/button.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import { ButtonType } from '../constants/button-types'
+
+interface Props extends React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement> {
+  variant: ButtonType
+  label: string
+}
+
+const Button: React.FC<Props> = (props: Props) => {
```

Comment:
já desestrutura suas props de um vez, você não precisa tipar o componente, Typescript é munido de inferência de tipos
```suggestion
const Button = ({variant, label}: Props) => {
```

Reply:
Esse ajuste entrou junto com outro tópico, então estão juntos no mesmo commit, segue ajustes:

commit: ec2e17718e44ed7c8cc7682a9919f1054d6d7ee2